common/src/main/java/com/linecorp/centraldogma/common/Entry.java

```diff
@@ -140,6 +158,14 @@ public Revision revision() {
         return revision;
     }
 
+    /**
+     * Returns a new {@link Entry} with the specified {@link Revision}.
+     */
+    public Entry<T> withRevision(Revision revision) {
+        requireNonNull(revision, "revision");
+        return new Entry<>(revision, path, type, content, true);
```

Comment:
Probably cache the `allowNullContent` and reuse it here for better accuracy?

Reply:
Makes sense.